spanner/spansql/parser.go

```diff
@@ -752,6 +753,22 @@ func (p *parser) skipSpace() bool {
 
 // advance moves the parser to the next token, which will be available in p.cur.
 func (p *parser) advance() {
+	// If the previous token was an identifier (quoted or unquoted),
+	// the next token being a dot means this is a path expression (not a number).
+	if p.cur.typ == quotedID || p.cur.typ == unquotedID {
+		// TODO: Is space around the dot in a path expression valid?
```

Comment:
Yes, whitespace characters (including linefeeds and tabs) are allowed both before and after the dot in a path expression.

Reply:
Okay, easy enough to implement, and made this a little simpler.